packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -383,6 +401,50 @@ export function TransferPanelMain({
     }
   }, [amount, setMaxAmount, setQueryParams])
 
+  useEffect(
+    // Show on page load if SC wallet since destination address mandatory
+    () => setShowAdvancedSettings(isSmartContractWallet),
+    [isSmartContractWallet]
+  )
+
+  useEffect(() => {
+    // Different destination address only allowed for tokens
+    if (!selectedToken) {
+      setDestinationAddress('')
```

Comment:
Shouldn't we have this at `undefined`?

Reply:
We could have these as `undefined`, but if used changes `destinationAddress` input to empty then the state value will be `''` regardless. So that in the example above we will have to check for `destinationAddress: destinationAddress || walletAddress` regardless